sky/clouds/service_catalog/oci_catalog.py

```diff
@@ -5,48 +5,86 @@
 
 History:
  - Hysun He (hysun.he@oracle.com) @ Apr, 2023: Initial implementation
+ - Hysun He (hysun.he@oracle.com) @ Jun, 2023: Reduce retry times by
+   excluding those unsubscribed regions.
 """
 
 import typing
 import logging
+import threading
 from typing import Dict, List, Optional, Tuple
+
+from pandas import DataFrame
 from sky.clouds.service_catalog import common
 from sky.skylet.providers.oci.config import oci_conf
+from sky.adaptors import oci as oci_adaptor
 
 if typing.TYPE_CHECKING:
     from sky.clouds import cloud
 
 logger = logging.getLogger(__name__)
 
-_df = common.read_catalog('oci/vms.csv')
+_df = None
 _image_df = common.read_catalog('oci/images.csv')
 
+__lock = threading.RLock()
```

Comment:
nit: we may want to use single `_` instead. reference: 
> Prepending a double underscore (__ aka “dunder”) to an instance variable or method effectively makes the variable or method private to its class (using name mangling); we discourage its use as it impacts readability and testability, and isn’t really private. Prefer a single underscore.

https://google.github.io/styleguide/pyguide.html#3162-naming-conventions

Reply:
ok